Sources/Events/StoredEvent.swift

```diff
@@ -16,17 +16,16 @@ import Foundation
 /// Contains the necessary information for storing and sending events.
 struct StoredEvent {
 
-    private(set) var encodedEvent: AnyEncodable
+    private(set) var encodedEvent: String
```

Comment:
Just for my understanding, does this change what the event looks like on disk? Wasn't it a string before?

Reply:
You're correct. It will be an encoded String though. Before it was an encoded dictionary. I think it's actually more clear now, without those `"_1":` and `"_0":`

Example of before:

```
{"user_id":"$RCAnonymousID:5b2741807e1b4d939767b3d2ed4fbe9a","event":{"impression":{"_0":{"date":"2024-11-25T20:30:42Z","id":"0B593A2F-1553-41DC-9CFD-16A12D89C418"},"_1":{"offering_identifier":"offering","locale_identifier":"en_ES","session_identifier":"2E57D4B8-55E3-4B24-93B1-55EA5F46DE38","display_mode":"full_screen","dark_mode":false,"paywall_revision":0}}}}
```

Example of after:

```
{"feature":"paywalls","event":"{\"impression\":{\"_1\":{\"paywall_revision\":0,\"session_identifier\":\"AF96931D-2A9F-4BA3-BBD2-DB255F2DCF0F\",\"display_mode\":\"full_screen\",\"offering_identifier\":\"offering\",\"locale_identifier\":\"en_ES\",\"dark_mode\":false},\"_0\":{\"date\":\"2024-11-26T10:20:23Z\",\"id\":\"BD9897AE-2C6D-4824-A9FE-7DBF20B8E313\"}}}","user_id":"$RCAnonymousID:5b2741807e1b4d939767b3d2ed4fbe9a"}
```

Reply:
Alright thanks! So does this mean we won't be able to decode whatever events were already stored on disk (encoded with the previous version)?

Reply:
I added a fallback here https://github.com/RevenueCat/purchases-ios/pull/4520/files/73db0703506e8ffb680907967b6012c2e3564942#diff-6b4517a9d0bd4d1dbde19c4fd19ce38bda2801d5b4c3c3fd3904cc30c2ec35bfR63 for that case

Reply:
Sweet, thanks for explaining!